tests/e2e/vm_connectivity_test.go

```diff
@@ -0,0 +1,205 @@
+package e2e
+
+import (
+	"fmt"
+	"github.com/deckhouse/virtualization/tests/e2e/executor"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	kc "github.com/deckhouse/virtualization/tests/e2e/kubectl"
+	virt "github.com/deckhouse/virtualization/tests/e2e/virtctl"
+)
+
+type Service struct {
+	ApiVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Metadata   struct {
+		Name      string `yaml:"name"`
+		Namespace string `yaml:"namespace"`
+	} `yaml:"metadata"`
+	Spec struct {
+		Ports []struct {
+			Port       int `yaml:"port"`
+			TargetPort int `yaml:"targetPort,omitempty"`
+			NodePort   int `yaml:"nodePort,omitempty"`
+		} `yaml:"ports"`
+	} `yaml:"spec"`
+}
+
+func getSVC(manifestPath string) *Service {
+	data, err := os.ReadFile(manifestPath)
+	if err != nil {
+		log.Fatalf("Error read file: %v", err)
+	}
+
+	var service Service
+	err = yaml.Unmarshal(data, &service)
+	if err != nil {
+		log.Fatalf("Error parsing yaml: %v", err)
+	}
+
+	return &service
+}
+
+var _ = Describe("VM connectivity", Ordered, ContinueOnFailure, func() {
+	imageManifest := vmPath("image.yaml")
+	vmOneManifest := vmPath("connectivity/vm1_connectivity_service.yaml")
+	vmTwoManifest := vmPath("connectivity/vm2_connectivity_service.yaml")
+	vmOneIPClaim := vmPath("connectivity/vm1_ipclaim.yaml")
+	vmTwoIPClaim := vmPath("connectivity/vm2_ipclaim.yaml")
+	vmSvcOne := vmPath("connectivity/vm1_svc.yaml")
+	vmSvcTwo := vmPath("connectivity/vm2_svc.yaml")
+
+	waitVmStatus := func(name, phase string) {
+		GinkgoHelper()
+		WaitResource(kc.ResourceVM, name, "jsonpath={.status.phase}="+phase, LongWaitDuration)
+	}
+
+	BeforeAll(func() {
+		By("Apply image for vms")
+		ApplyFromFile(imageManifest)
+		WaitFromFile(imageManifest, PhaseReady, LongWaitDuration)
+	})
+
+	AfterAll(func() {
+		By("Delete all manifests")
+		files := make([]string, 0)
+		err := filepath.Walk(
+			conf.VM.TestDataDir, func(path string, info fs.FileInfo, err error) error {
+				if err == nil && strings.HasSuffix(info.Name(), "yaml") {
+					files = append(files, path)
+				}
+				return nil
+			},
+		)
+		if err != nil || len(files) == 0 {
+			kubectl.Delete(imageManifest, kc.DeleteOptions{})
+			kubectl.Delete(conf.VM.TestDataDir, kc.DeleteOptions{})
+		} else {
+			for _, f := range files {
+				kubectl.Delete(f, kc.DeleteOptions{})
+			}
+		}
+	})
+
+	Context("Connectivity test", func() {
+		CheckResultSshCommand := func(vmName, command, equal string) {
+			GinkgoHelper()
+			res := virtctl.SshCommand(vmName, command, virt.SshOptions{
+				Namespace:   conf.Namespace,
+				Username:    "cloud",
+				IdenityFile: vmPath("sshkeys/id_ed"),
```

Comment:
how did you solve the problem with the git rules? the key must have a mode 600

Reply:
Add func ChmodFile